src/App.tsx

```diff
@@ -1,61 +1,117 @@
+import { useState } from 'react';
+import type { ChangeEvent, FormEvent } from 'react';
+
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+
+import type { User } from './types/User';
+import type { Todo } from './types/Todo';
+
+import { TodoList } from './components/TodoList/TodoList';
+
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find((user) => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map((todo) => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [userId, setUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [isUser, setIsUser] = useState(true);
+  const [isTitle, setIsTitle] = useState(true);
+  const [todosList, setTodosList] = useState(todos);
+
+  const clearForm = () => {
+    setTitle('');
+    setUserId(0);
+
+    setIsTitle(true);
+    setIsUser(true);
+  };
+
+  const selectUser = (event: ChangeEvent<HTMLSelectElement>) => {
+    setUserId(Number(event.target.value));
+    setIsUser(true);
+  };
+
+  const changeTitle = (event: ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitle(true);
+  };
+
+  const addTodo = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setIsTitle(Boolean(title));
+    setIsUser(Boolean(userId));
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: todos.sort((a, b) => b.id - a.id)[0].id + 1,
```

Comment:
a bit cumbersome construct
i'd rather go with .map + Math.max
```suggestion
      id: Math.max(...todos.map(todo => todo.id)) + 1,
```

Reply:
or make a tiny helper to just make it look more neat